lib/ldp/client/methods.rb

```diff
@@ -69,7 +69,7 @@ def post url, body = nil, headers = {}
     logger.debug "LDP: POST [#{url}]"
     resp = http.post do |req|
       req.url munge_to_relative_url(url)
-      req.headers = default_headers.merge headers
+      req.headers = req.headers.merge!(default_headers).merge(headers)
```

Comment:
I think @cbeer was suggesting something like:
```ruby 
req.headers.merge!(default_headers).merge!(headers)
```
Then you can do away with the assignment.